### PR TITLE
[proof system] Reducing use of the term `plonk` 

### LINKED ISSUE #922: Reduce overloading of `plonk` term in cocdebase

### PR DESCRIPTION
**Pull request by pdg744**
The term `plonk` is overloaded and confusing in our codebase; this PR reduces use of that term to disambiguate meaning. 

Please review based on whether the suggested names make sense; we can worry about the implications in terms of breaking changes once we're in alignment about whether the names are good names. 

Resolves #922

### CODE REVIEW

File: risc0/zkvm/src/host/server/prove/exec.rs
Comment (tzerrell) [inline risc0/zkvm/src/host/server/prove/exec.rs:249]: Someone closer to the circuit code than me should check whether the various places you switched "RAM" to "Memory" in this PR are ok -- it's possible the "Random Access" portion of the term is load-bearing.

**Review by tzerrell — COMMENTED** [review]
Nice, reducing our dependence on people understanding which meaning of "PLONK" we have in mind is definitely a good thing!

One inline comment that needs another reviewer's attention.

File: risc0/zkvm/src/host/server/prove/exec.rs
Comment (pdg744) [reply risc0/zkvm/src/host/server/prove/exec.rs:249]: oh good call -- i don't have any opinion between, e.g., `ram_table` vs. `memory_table`

File: risc0/zkp/src/prove/accum.rs
Comment (flaub) [inline risc0/zkp/src/prove/accum.rs:104]: Unfortunately we can't change these until we make a corresponding change in the circuit.

File: risc0/zkvm/src/host/server/prove/exec.rs
Comment (flaub) [reply risc0/zkvm/src/host/server/prove/exec.rs:249]: Many of these names have corresponding components in the circuit. In some places it's required that the names match (like in the case of extern calls that are made from the circuit out to the host). In other cases, it would be confusing to a circuit dev.

I really like the idea of renaming things, but we need to make the circuit changes first.

**Review by tzerrell — CHANGES_REQUESTED** [review]
See the other comments -- this is blocked until we can make corresponding naming changes on the circuit.

I'm glad we're figuring out what we want these names to be, though, so we are ready to change them both here & in the circuit when we have the opportunity.

File: risc0/zkvm/src/host/server/prove/exec.rs
Comment (pdg744) [reply risc0/zkvm/src/host/server/prove/exec.rs:249]: Yeah I figured many of the changes I propose here will necessitate other changes in order to be coherent; I guess first we can sort out whether the suggestions I make here actually make sense, and if so, we can make a plan for how to get all the relevant dependencies sorted.

Comment (flaub) [reply risc0/zkvm/src/host/server/prove/exec.rs:249]: I'm partial to `ram` rather than `memory` since `ram` is well known in CS and actually is more specific than memory.

**Comment by pdg744** [comment]
Added this note to the PR description: 

*Please review based on whether the suggested names make sense; we can worry about the implications in terms of breaking changes once we're in alignment about whether the names are good names.*

**Comment by tzerrell** [comment]
> Added this note to the PR description:
> 
> _Please review based on whether the suggested names make sense; we can worry about the implications in terms of breaking changes once we're in alignment about whether the names are good names._

I'm going to drop my request for changes, because I like these names with no objections beyond what's in already-seen comments, and CI will prevent us from accidentally landing this without making the corresponding circuit changes.

File: risc0/zkvm/src/host/server/prove/exec.rs
Comment (pdg744) [inline risc0/zkvm/src/host/server/prove/exec.rs:65]:
```suggestion
    pub ram_table: plonk::RamTable,
```

File: risc0/zkp/src/prove/accum.rs
Comment (pdg744) [reply risc0/zkp/src/prove/accum.rs:104]: Marking this as resolved in order to focus attention on the question of whether the naming suggestion improves clarity; will worry about breaking changes later

File: risc0/zkvm/src/host/server/prove/exec.rs
Comment (pdg744) [reply risc0/zkvm/src/host/server/prove/exec.rs]: Updated to avoid making any changes `ram` and `memory`.

**Comment by pdg744** [comment]
OK I've updated the PR to avoid making any changes from `ram` to `memory`. 

@jbruestle how do the terms here feel to you?

**Comment by pdg744** [comment]
Discussed this with proof system team at October 16th sync -- closing this for now. We can revisit this again in the future as appropriate, perhaps in conjunction with audits and/or implementing LogUp

**Pull request closed without merging**
